src/shared/components/blast-sequence-button/BlastSequenceButton.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { useNavigate } from 'react-router';
+import classNames from 'classnames';
+
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import useBlastForm from 'src/content/app/tools/blast/hooks/useBlastForm';
+
+import { useAppDispatch } from 'src/store';
+
+import {
+  addSelectedSpecies,
+  clearBlastForm
+} from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+
+import { ImageButton } from 'src/shared/components/image-button/ImageButton';
+import { BlastIcon } from 'src/shared/components/app-icon';
+
+import type { Species } from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+import type { SequenceType } from 'src/content/app/tools/blast/types/blastSettings';
+
+import styles from './BlastSequenceButton.scss';
+
+// TODO: think about the possible header for the sequence
+type Props = {
+  label?: string;
+  species: Species;
+  sequence?: string; // if the sequence isn't provided, the button will be disabled
+  sequenceType: SequenceType;
+  className?: string;
+};
+
+const BlastSequenceButton = (props: Props) => {
+  const defaultLabel = 'Blast whole sequuence';
+  const {
+    label = defaultLabel,
+    species,
+    sequence,
+    sequenceType,
+    className
+  } = props;
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const { setSequenceForGenome } = useBlastForm();
+
+  const onClick = () => {
+    if (!sequence) {
+      return;
+    }
+
+    dispatch(clearBlastForm());
+    setSequenceForGenome({
+      sequence: { value: sequence },
+      species,
+      sequenceType
+    });
+    dispatch(addSelectedSpecies(species));
+
+    navigate(urlFor.blastForm());
+  };
+
+  const componentClasses = classNames(styles.container, className);
```

Comment:
`componentClasses` -> `containerClasses` ?

Reply:
I don't know. I was trying to express the idea that this is the topmost class that applies to the whole component.

Reply:
To me componentClasses are the one we see in the dom when inspecting it with the actual react component name so for this one something like `_BlastSequenceButton_` :shrug:

Reply:
I couldn't find another alternative to componentClasses so happy to go with it.